src/OpenTelemetry/Batch.cs

```diff
@@ -110,8 +141,9 @@ public void Dispose()
             public bool MoveNext()
             {
                 var circularBuffer = this.circularBuffer;
+                var metrics = this.metrics;
 
-                if (circularBuffer == null)
+                if (circularBuffer == null && metrics == null)
```

Comment:
I'm slightly concerned about the extra perf cost although it might be just 2 nanoseconds, as this will contribute to every single export call. https://github.com/open-telemetry/opentelemetry-dotnet/pull/2295/files#r700611706

Reply:
See the final decision here: https://github.com/open-telemetry/opentelemetry-dotnet/pull/2327/files#r704879293